week-2/mandatory/2-ecommerce-db/task.md

```diff
@@ -7,8 +7,79 @@ In this homework, you are going to work with an ecommerce database. In this data
+
+3. SELECT * FROM products WHERE product_name LIKE '%socks%';
+
+4. SELECT products.id, products.product_name, pa.unit_price, pa.supp_id
+    FROM products JOIN product_availability pa ON products.id=pa.prod_id
+    WHERE pa.unit_price > 100;
+
+5. SELECT products.product_name most_expensive_products, pa.unit_price
+    FROM products JOIN product_availability pa ON products.id=pa.prod_id
+    ORDER BY pa.unit_price DESC
+    LIMIT 5;
+
+6. SELECT products.product_name, pa.unit_price, sup.supplier_name FROM
+    products JOIN product_availability pa ON
+    (products.id=pa.prod_id) JOIN suppliers sup ON
+    (pa.supp_id=sup.id);
+
+7. SELECT p.product_name, sup.supplier_name FROM
+    products p JOIN product_availability pa ON
+    (p.id=pa.prod_id) JOIN suppliers sup ON
+    (pa.supp_id=sup.id) WHERE sup.country='United Kingdom';
+
+8. SELECT cust.id cust_id, orders.id order_id, orders.order_reference,
+    orders.order_date, oi.quantity * pa.unit_price as total_cost
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN product_availability pa ON (oi.supplier_id=pa.supp_id)
+    WHERE cust.id=1;
+
+9.
+    SELECT cust.id cust_id, cust.name cust_name, orders.id order_id, orders.order_date,
+    orders.order_reference, oi.product_id, oi.supplier_id, oi.quantity
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    WHERE cust.name='Hope Crosby';
+
+ -- if i select orders.* and oi.* list duplicate unnecessary id numbers and hard to read, so tailored by selecting usefull fields..stingy customer btw--
+
+ 10.
+    SELECT pro.product_name, pa.unit_price, oi.quantity, orders.order_reference
+    FROM products pro JOIN product_availability pa ON (pro.id=pa.prod_id)
+    JOIN order_items oi ON (pa.prod_id=oi.product_id)
+    JOIN orders ON (oi.order_id=orders.id)
+    WHERE orders.order_reference='ORD006';
+
+11.
+    SELECT cust.name cust_name, orders.order_reference, orders.order_date,
+    pro.product_name, sup.supplier_name, oi.quantity
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN products pro ON (oi.product_id=pro.id)
+    JOIN suppliers sup ON (oi.supplier_id=sup.id);
+
+12.
+    SELECT DISTINCT cust.name suspected_customers
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN suppliers sup ON (oi.supplier_id=sup.id)
+    WHERE sup.country='China';
+
+13.
+    SELECT cust.name cust_names, orders.order_reference, orders.order_date,
+    oi.quantity * pa.unit_price total_amount
+    FROM customers cust JOIN orders ON (cust.id=orders.customer_id)
+    JOIN order_items oi ON (orders.id=oi.order_id)
+    JOIN product_availability pa ON (oi.product_id=pa.prod_id)
+    ORDER BY total_amount DESC;
 
```

Comment:
missing grouping here, as it looks for aggregating the value of total

Reply:
Groups added